Client/Frontend/Browser/BrowserLongPressHandler.swift

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import UIKit
+
+enum ElementType {
+
+    case ImageSrc(url:NSURL,location:CGRect)
+    case hrefLink(url:NSURL,location:CGRect)
+}
+
+protocol BrowserLongPressHandlerDelegate {
+    func webView(webview: WKWebView, elementsDidLongPressed elements: [ElementType])
+}
+
+class BrowserLongPressHandler: NSObject, UIGestureRecognizerDelegate {
+
+    let webView: WKWebView?
+    var longPressHandlerDelegate: BrowserLongPressHandlerDelegate?
+    let longPressGesture: UILongPressGestureRecognizer?
+
+    init(webView: WKWebView) {
+        super.init()
+        self.webView = webView
+        longPressGesture = UILongPressGestureRecognizer(target: self, action: "didLongPress:")
+        longPressGesture?.delegate = self
+        longPressGesture?.minimumPressDuration *= 0.9
+        self.webView!.addGestureRecognizer(longPressGesture!)
+        var error: NSError?
+        let javaScript = String(contentsOfFile: NSBundle.mainBundle().pathForResource("LongPress", ofType: "js")!, encoding: NSUTF8StringEncoding, error: &error)!
+
+        let longPressUserScript = WKUserScript(source:javaScript, injectionTime: WKUserScriptInjectionTime.AtDocumentStart, forMainFrameOnly: false)
+        self.webView!.configuration.userContentController.addUserScript(longPressUserScript)
+    }
+
+
+    // MARK: - Gesture Recognizer Delegate Methods
+    func gestureRecognizer(gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWithGestureRecognizer otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return true
+    }
+
+    // MARK: - Long Press Gesture Handling
+    func didLongPress(gestureRecognizer: UILongPressGestureRecognizer) {
+        if gestureRecognizer.state == UIGestureRecognizerState.Began {
+            //Finding actual touch location in webView
+            var touchLocation = gestureRecognizer.locationInView(webView)
+            touchLocation.x -= webView!.scrollView.contentInset.left
+            touchLocation.y -= webView!.scrollView.contentInset.top
+            touchLocation.x /= webView!.scrollView.zoomScale
+            touchLocation.y /= webView!.scrollView.zoomScale
+
+            evaluateTouchLocation(touchLocation, gestureRecognizer: gestureRecognizer)
+        }
+    }
+
+    ///Check whether long pressed is on valid link. If found valid link then find the rect for link.
+    func evaluateTouchLocation(touchLocation: CGPoint,gestureRecognizer: UILongPressGestureRecognizer) {
```

Comment:
Make this private (along with all other methods below that aren't called outside of this class).